cocos/core/gfx/webgl/webgl-commands.ts

```diff
@@ -2775,8 +2776,8 @@ export function WebGLCmdFuncCopyBuffersToTexture (
             stride.width = region.buffStride > 0 ?  region.buffStride : extent.width;
             stride.height = region.buffTexHeight > 0 ? region.buffTexHeight : extent.height;
 
-            const destWidth  = region.texExtent.width + offset.x === gpuTexture.width ? region.texExtent.width : extent.width;
-            const destHeight = region.texExtent.height + offset.y === gpuTexture.height ? region.texExtent.height : extent.height;
+            const destWidth  = region.texExtent.width + offset.x === gpuTexture.width >> mipLevel ? region.texExtent.width : extent.width;
+            const destHeight = region.texExtent.height + offset.y === gpuTexture.height >> mipLevel ? region.texExtent.height : extent.height;
```

Comment:
We can add parenthesis to help understanding. ` === ((gpuTexture.width >> mipLevel) ? region.texExtent.width : extent.width);`

Reply:
resolved